### PR TITLE
Add signal tools page to documentation

### LINKED ISSUE #171: Find maximum and width of a peak
Similar to the `centroid` function from #168 we should create further functions that give back information about a peak without fitting

1. Maximum of a peak
HyperSpy provides the `valuemax` function to find the position of a maximum of a signal. With smoothing and signal ranges that can work also on noisier data or data with multiple peaks, respectively. However,  an alternative approach that I would like to see implemented is taking the first derivative and then finding zero-crossings. In particular this method is useful to identify the positions of secondary maxima.
EDIT: This functionality is actually provided by [find_peaks1D_ohaver](https://hyperspy.readthedocs.io/en/latest/api/hyperspy._signals.signal1d.html#hyperspy._signals.signal1d.Signal1D.find_peaks1D_ohaver)  -- its been a while since I had looked at that function

2. Width of a peak
From fits, you can get the FWHM of a peak, but that is not very helpful for asymmetric peak shapes. I therefore envisage a function that calculates the distance between the half maximum points of a peak along the signal axis.
EDIT: This is actually provided by [estimate_peak_width](https://hyperspy.readthedocs.io/en/latest/api/hyperspy._signals.signal1d.html#hyperspy._signals.signal1d.Signal1D.find_peaks1D_ohaver)

### PR DESCRIPTION
**Pull request by jlaehne**
### Requirements
Split `signal tools` page out of the `utilities page in the user guide. Add some important HyperSpy functions.
Some additional small improvements to the documentation.

Closes #171.

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update user guide (if appropriate),
- [x] ready for review.

### CODE REVIEW

File: doc/source/user_guide/signal_tools.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    >>> peaks = s.find_peaks1D_ohaver()
+    >>> peaks = s.isig[100:-100].find_peaks1D_ohaver()
+
+.. _peak_width:
+
+Peak Width
+----------
+
+For asymmetric peaks, `fitted functions <fitting_luminescence>` may not provide
```
Comment (jordiferrero) [inline doc/source/user_guide/signal_tools.rst:82]: This is not parsed correctly.

File: doc/source/user_guide/signal_tools.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Peak positions and properties
+=============================
+
+.. _find_peaks:
+
+Peak identification
+-------------------
+
+HyperSpy provides functions to finde the positions of maxima or minima in a
```
Comment (jordiferrero) [inline doc/source/user_guide/signal_tools.rst:53]: *find, not finde (spelling mistake)

**Review by jordiferrero — COMMENTED** [review]
I like the documentation like this much more as the utils section was a bit confusing.

**Comment by jlaehne** [comment]
> I like the documentation like this much more as the utils section was a bit confusing.

I agree, I'll try to extend and improve it a bit further the next days.

**Review by jordiferrero — APPROVED** [review]

**Pull request merged**